tests/functions_tests/test_basic_math.py

```diff
@@ -157,6 +157,95 @@ def test_rdiv_backward_gpu(self): self.backward_gpu(lambda x, y: y / x)
+    def test_sub_forward_cpu(self):  self.forward_cpu(lambda x, y: x - y)
+    # def test_rsub_forward_cpu(self): self.forward_cpu(lambda x, y: y - x)
+    def test_mul_forward_cpu(self):  self.forward_cpu(lambda x, y: x * y)
+    # def test_rmul_forward_cpu(self): self.forward_cpu(lambda x, y: y * x)
+    def test_div_forward_cpu(self):  self.forward_cpu(lambda x, y: x / y)
+    # def test_rdiv_forward_cpu(self): self.forward_cpu(lambda x, y: y / x)
+    def test_pow_forward_cpu(self):  self.forward_cpu(lambda x, y: x ** y)
+    # def test_rpow_forward_cpu(self): self.forward_cpu(lambda x, y: y ** x)
+
+    def forward_gpu(self, op):
+        self.check_forward(op, to_gpu(self.x), True)
+
+    def test_add_forward_gpu(self):  self.forward_gpu(lambda x, y: x + y)
+    # def test_radd_forward_gpu(self): self.forward_gpu(lambda x, y: y + x)
+    def test_sub_forward_gpu(self):  self.forward_gpu(lambda x, y: x - y)
+    # def test_rsub_forward_gpu(self): self.forward_gpu(lambda x, y: y - x)
+    def test_mul_forward_gpu(self):  self.forward_gpu(lambda x, y: x * y)
+    # def test_rmul_forward_gpu(self): self.forward_gpu(lambda x, y: y * x)
+    def test_div_forward_gpu(self):  self.forward_gpu(lambda x, y: x / y)
+    # def test_rdiv_forward_gpu(self): self.forward_gpu(lambda x, y: y / x)
+    def test_pow_forward_gpu(self):  self.forward_gpu(lambda x, y: x ** y)
+    # def test_rpow_forward_gpu(self): self.forward_gpu(lambda x, y: y ** x)
+
+    def check_backward(self, op, x_data, y_grad, gpu):
+        value = self.value
+        if gpu:
+            value = cuda.to_gpu(value)
+        x = Variable(x_data)
+        y = op(x, value)
+        y.grad = y_grad
+        y.backward()
+
+        func = y.creator
+        f = lambda: func.forward((x.data,))
+        gx, = numerical_grad(f, (x.data,), (y.grad,))
+
+        assert_allclose(gx, x.grad, atol=1e-4, rtol=1e-4)
+
+    def backward_cpu(self, op):
+        self.check_backward(op, self.x, self.gy, False)
+
+    def test_add_backward_cpu(self):  self.backward_cpu(lambda x, y: x + y)
+    # def test_radd_backward_cpu(self): self.backward_cpu(lambda x, y: y + x)
+    def test_sub_backward_cpu(self):  self.backward_cpu(lambda x, y: x - y)
+    # def test_rsub_backward_cpu(self): self.backward_cpu(lambda x, y: y - x)
+    def test_mul_backward_cpu(self):  self.backward_cpu(lambda x, y: x * y)
+    # def test_rmul_backward_cpu(self): self.backward_cpu(lambda x, y: y * x)
+    def test_div_backward_cpu(self):  self.backward_cpu(lambda x, y: x / y)
+    # def test_rdiv_backward_cpu(self): self.backward_cpu(lambda x, y: y / x)
+    def test_pow_backward_cpu(self):  self.backward_cpu(lambda x, y: x ** y)
+    # def test_rpow_backward_cpu(self): self.backward_cpu(lambda x, y: y ** x)
+
+    def backward_gpu(self, op):
+        self.check_backward(op, to_gpu(self.x), to_gpu(self.gy), True)
+
+    # 2015/6/15 Kenta OONO <oono@preferred.jp>
```

Comment:
Remove date and author information from regular comments. They can be extracted from the repository log.